modules/openapi-generator/src/main/resources/python/model_anyof.mustache

```diff
@@ -97,13 +101,13 @@ class {{classname}}({{#parent}}{{{.}}}{{/parent}}{{^parent}}BaseModel{{/parent}}
             return v
 
     @classmethod
-    def from_dict(cls, obj: dict) -> {{{classname}}}:
+    def from_dict(cls, obj: dict) -> Self:
         return cls.from_json(json.dumps(obj))
 
     @classmethod
-    def from_json(cls, json_str: str) -> {{{classname}}}:
+    def from_json(cls, json_str: str) -> Self:
```

Comment:
I'm ok with the change but just want to confirm one thing. IDE should be able to show the actual class name (e.g Pet) instead of `Self`, right?

Reply:
This will depend on the IDE.

FWIW, VS Code translates `Self` to the related class name:

![Screenshot from 2023-10-01 17-28-00](https://github.com/OpenAPITools/openapi-generator/assets/65311/ed0ccd5d-3789-40d5-a7b6-f4f68ba5990b)
![Screenshot from 2023-10-01 17-28-06](https://github.com/OpenAPITools/openapi-generator/assets/65311/e5b1c9db-84d2-4670-acd6-65def938ee7f)

Mypy also correctly resolves the name (as seen by `reveal_type` on `reveal_type(xx)`):

```
foobar.py:9: note: Revealed type is "petstore_api.models.pet.Pet"
```

Reply:
Nice. Just want to make sure most IDE can handle it.